app/src/androidTest/java/com/example/sharingang/MapActivityTest.kt

```diff
@@ -27,29 +25,28 @@ class MapActivityTest {
     private fun grantPermission() {
         val instrumentation = InstrumentationRegistry.getInstrumentation()
         if (Build.VERSION.SDK_INT >= 23) {
-            val allowPermission = UiDevice.getInstance(instrumentation).findObject(UiSelector().text(
+            val allowPermission = UiDevice.getInstance(instrumentation).findObject(
+                UiSelector().text(
                     when {
                         Build.VERSION.SDK_INT == 23 -> "Allow"
                         Build.VERSION.SDK_INT <= 28 -> "ALLOW"
                         Build.VERSION.SDK_INT == 29 -> "Allow only while using the app"
                         else -> "While using the app"
                     }
-            ))
+                )
+            )
             if (allowPermission.exists()) {
                 allowPermission.click()
             }
         }
     }
 
     @Test
-    fun clickingOnButtonDisplaysLocation() {
-        val button = onView(withId(R.id.button_get_location))
-        button.check(matches(withText("Get Location")))
+    fun grantingPermissionDisplaysLocation() {
+        grantPermission()
         val text = onView(withId(R.id.location_display))
         text.check(matches(withText("")))
-
-        button.perform(click())
-        grantPermission()
-        button.perform(click())
+        Thread.sleep(10000)
+        text.check(matches(withText(containsString("Your location"))))
```

Comment:
Does anyone know a way to wait for the text to change without sleeping?
I've found something using IdleResources but it requires changes in the class specifically for the test which it not very good.